### PR TITLE
RememberMe  #537

### LINKED ISSUE #537: RememberMe
As a user, I want my information to be filled up after I check the remember me box

[Story Points: 2]
[Priority: low]
[Risk: low]

### PR DESCRIPTION
**Pull request by eyadsw**
closes #537

### CODE REVIEW

**Review by JamesPartsafas — COMMENTED** [review]
I think there was a miscommunication for this feature. Remember me functions usually mean when a user closes their browser or tab and then goes back to the site, they remain logged in. If a user does not have remember me enabled, when they close their browser or tab, they are automatically logged out and if they go back to the site, they'll be told to login again. Currently this does not do this, but instead simply stores the last email and password used in plaintext to autofill the login forms when the user logs out (which is a security risk).

Remember me functionality can be achieved by saving the JWT to local storage instead of session storage. If the user does not want remember me enabled, the JWT should instead be saved to session storage instead of local storage.

Remember me should be true by default, instead of false as you've set it now.

The register page also has a remember me option, it's not only the login page as you've done here.

**Comment by eyadsw** [comment]
> I think there was a miscommunication for this feature. Remember me functions usually mean when a user closes their browser or tab and then goes back to the site, they remain logged in. If a user does not have remember me enabled, when they close their browser or tab, they are automatically logged out and if they go back to the site, they'll be told to login again. Currently this does not do this, but instead simply stores the last email and password used in plaintext to autofill the login forms when the user logs out (which is a security risk).
> 
> Remember me functionality can be achieved by saving the JWT to local storage instead of session storage. If the user does not want remember me enabled, the JWT should instead be saved to session storage instead of local storage.
> 
> Remember me should be true by default, instead of false as you've set it now.
> 
> The register page also has a remember me option, it's not only the login page as you've done here.

fixed it as you mentioned

**Review by JamesPartsafas — COMMENTED** [review]
This is improved in that when remember me is not checked, then closing the tab and going back to the site has you logged out, but the rest of the site doesn't work when logged in with remember me unchecked (for example, you cannot create dossiers, manage your groups, or even log out). I may be wrong, but I suspect modifying the initializeUser function in App.tsx to check for both local and session storage might fix the issue.

**Comment by eyadsw** [comment]
> This is improved in that when remember me is not checked, then closing the tab and going back to the site has you logged out, but the rest of the site doesn't work when logged in with remember me unchecked (for example, you cannot create dossiers, manage your groups, or even log out). I may be wrong, but I suspect modifying the initializeUser function in App.tsx to check for both local and session storage might fix the issue.

you were right I had to give authorization to users logged in with remember me unchecked. now the users/admins are able to interact with the website regularly.

File: App/src/services/auth.ts [hunk 1]
```diff
@@ -53,9 +51,19 @@ export function RegisterUser(dto: RegisterDTO): Promise<AuthenticationResponse>
...
-    return axios.post("/Authentication/Logout").then(() => {
-        //remove token from local storage
-        localStorage.removeItem("token");
+    return new Promise<void>((resolve, reject) => {
+        axios
+            .post("/Authentication/Logout")
+            .then(() => {
+                // Clear the token and authorization header
+                localStorage.removeItem("token");
+                localStorage.removeItem("accessToken");
```
Comment (JamesPartsafas) [inline App/src/services/auth.ts:60]: I think you meant to remove this from sessionStorage, but otherwise looks good

**Review by JamesPartsafas — APPROVED** [review]

**Pull request merged**